[Bug] umi 3.3.5 "npm run start" 启动后，Ctrl-C 关闭进程时报错

<!--
⚠️ ⚠️ ⚠️ 注意：讨论和提问请到讨论区（https://github.com/umijs/umi/discussions），否则会被直接关掉。 ⚠️ ⚠️ ⚠️
-->
<!--
感谢您向我们反馈问题，为了高效的解决问题，我们期望你能提供以下信息：
-->

## What happens?

Umi 3.3.5 （本机测了下，从 3.2.21 往后所有版本都有这个问题）通过 package.json 中的 "npm run start" 启动后，Ctrl-C 关闭进程时报错。

## Mini Showcase Repository(REQUIRED)

复现过程：

① 在空目录中（比如 "ttt"）新建 package.json，内容如下：

package.json 内容：

```
{
  "name": "ttt",
  "version": "1.0.0",
  "description": "", 
  "main": "index.js",
  "private": true,
  "scripts": {
    "start":"umi dev",
    "test": "echo \"Error: no test specified\" && exit 1"
  },  
  "author": "", 
  "license": "ISC",
  "dependencies": {
    "umi": "^3.3.5"                                                                                                                                                               
  }
}
```

② 当前目录执行 "tnpm i"（阿里内网）
③ 执行 "npm run start" 启动 umi
执行到这里尚一切正常
④ Ctrl-C 结束进程，这时报错，内容如下：

```
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ttt@1.0.0 start: `umi dev`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the ttt@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/bachi/.npm/_logs/2021-01-16T14_25_32_898Z-debug.log
```

其中错误日志 "/Users/bachi/.npm/_logs/2021-01-16T14_25_32_898Z-debug.log"文件内容如下：

```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'run', 'start' ]
2 info using npm@6.14.2
3 info using node@v10.15.3
4 verbose run-script [ 'prestart', 'start', 'poststart' ]
5 info lifecycle ttt@1.0.0~prestart: ttt@1.0.0
6 info lifecycle ttt@1.0.0~start: ttt@1.0.0
7 verbose lifecycle ttt@1.0.0~start: unsafe-perm in lifecycle true
8 verbose lifecycle ttt@1.0.0~start: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/Users/bachi/gitlab/ttt/node_modules/.bin:/usr/local/bin/:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/Users/bachi/android/alipay-adt/sdk/platform-tools:/Users/bachi/android/alipay-adt/sdk/tools:/Users/bachi/flutter/bin:/Users/bachi/android/alipay-adt/sdk:/Users/bachi/android/apache-maven-3.0.5/bin:/usr/local/lib/node_modules/jsbeautify:/Users/bachi/.rvm/bin:/usr/local/go/bin:/Users/bachi/go/bin:/Users/bachi/apache-maven-3.6.3/bin:/Users/bachi/bin
9 verbose lifecycle ttt@1.0.0~start: CWD: /Users/bachi/gitlab/ttt
10 silly lifecycle ttt@1.0.0~start: Args: [ '-c', 'umi dev' ]
11 silly lifecycle ttt@1.0.0~start: Returned: code: 1  signal: null
12 info lifecycle ttt@1.0.0~start: Failed to exec start script
13 verbose stack Error: ttt@1.0.0 start: `umi dev`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:189:13)
13 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:189:13)
13 verbose stack     at maybeClose (internal/child_process.js:970:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:259:5)
14 verbose pkgid ttt@1.0.0
15 verbose cwd /Users/bachi/gitlab/ttt
16 verbose Darwin 19.3.0
17 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "start"
18 verbose node v10.15.3
19 verbose npm  v6.14.2
20 error code ELIFECYCLE
21 error errno 1
22 error ttt@1.0.0 start: `umi dev`
22 error Exit status 1
23 error Failed at the ttt@1.0.0 start script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

如果把 package.json 中的 umi 版本从 3.3.5 降级为 3.2.20 ，重复上述过程则不会报错

```
"dependencies": {
   "umi": "^3.3.5"   →    "3.2.20"                                                                                                                                                            
}
```

> umi 3.3.5 版本下，当前目录直接运行 umi dev 不报错："npx umi dev"，Ctrl-C 正常无错退出。
> 从包括 umi 3.2.21 在内的往后所有版本都有这个问题。


## Context

- **Umi Version**: 本机没有全局安装 umi
- **Node Version**: v10.15.3
- **Platform**: MacBook Pro，"uname -a":  Darwin bogon 19.3.0 Darwin Kernel Version 19.3.0: Thu Jan  9 20:58:23 PST 2020; root:xnu-6153.81.5~1/RELEASE_X86_64 x86_64

--- a/docs/config/README.md
+++ b/docs/config/README.md
@@ -1027,7 +1027,7 @@ export default defineConfig({
 
 配置路由。
 
-umi 的路由基于 [react-router@5](https://reacttraining.com/react-router/web/guides/quick-start) 实现，配置和 react-router 基本一致，详见[路由配置](TODO)章节。
+umi 的路由基于 [react-router@5](https://reacttraining.com/react-router/web/guides/quick-start) 实现，配置和 react-router 基本一致，详见[路由配置](./docs/routing)章节。
 
 比如：
 

--- a/docs/config/README.zh-CN.md
+++ b/docs/config/README.zh-CN.md
@@ -1025,7 +1025,7 @@ export default defineConfig({
 
 配置路由。
 
-umi 的路由基于 [react-router@5](https://reacttraining.com/react-router/web/guides/quick-start) 实现，配置和 react-router 基本一致，详见[路由配置](TODO)章节。
+umi 的路由基于 [react-router@5](https://reacttraining.com/react-router/web/guides/quick-start) 实现，配置和 react-router 基本一致，详见[路由配置](./docs/routing)章节。
 
 比如：
 

--- a/packages/umi/src/cli.ts
+++ b/packages/umi/src/cli.ts
@@ -34,9 +34,12 @@ if (args.version && !args._[0]) {
         });
         // ref:
         // http://nodejs.cn/api/process/signal_events.html
+        // https://lisk.io/blog/development/why-we-stopped-using-npm-start-child-processes
         process.on('SIGINT', () => {
           child.kill('SIGINT');
-          process.exit(1);
+          // ref:
+          // https://github.com/umijs/umi/issues/6009
+          process.exit(0);
         });
         process.on('SIGTERM', () => {
           child.kill('SIGTERM');

